### ISSUE TITLE
GitLab: skip projects with repo disabled

### ISSUE THREAD

**Issue by arsdehnel** [renovatebot/renovate#5080]
**What would you like Renovate to be able to do?**

For my self-hosted GitLab I have a lot of projects that have the repo disabled.  This is exposed in the `/projects` API response as `repository_access_level: "disabled"`.   Right now I get a fatal platform exception on these:

```
 INFO: Initialising git repository into /var/folders/1f/xpfn1tys6hg101s6l55hw_5r0001qs/T/renovate/repos/gitlab/groupName/projectName (repository=groupName/projectName)
DEBUG: git clone error (repository=groupName/projectName)
       "err": {
         "message": "Cloning into '.'...\nremote: You are not allowed to download code from this project.\nfatal: unable to access 'https://**redacted**@gitlab.example.com/groupName/projectName.git/': The requested URL returned error: 403\n",
         "stack": "Error: Cloning into '.'...\nremote: You are not allowed to download code from this project.\nfatal: unable to access 'https://**redacted**@gitlab.example.com/groupName/projectName.git/': The requested URL returned error: 403\n\n    at /Users/foo/.nvm/versions/node/v10.13.0/lib/node_modules/renovate/node_modules/simple-git/promise.js:62:29\n    at Git.<anonymous> (/Users/foo/.nvm/versions/node/v10.13.0/lib/node_modules/renovate/node_modules/simple-git/src/git.js:214:18)\n    at Function.Git.fail (/Users/foo/.nvm/versions/node/v10.13.0/lib/node_modules/renovate/node_modules/simple-git/src/git.js:1474:18)\n    at fail (/Users/foo/.nvm/versions/node/v10.13.0/lib/node_modules/renovate/node_modules/simple-git/src/git.js:1432:20)\n    at /Users/foo/.nvm/versions/node/v10.13.0/lib/node_modules/renovate/node_modules/simple-git/src/git.js:1441:16\n    at process._tickCallback (internal/process/next_tick.js:68:7)"
       }
DEBUG: Caught initRepo error (repository=groupName/projectName)
       "err": {
         "message": "platform-failure",
         "stack": "Error: platform-failure\n    at Storage.initRepo (/Users/foo/.nvm/versions/node/v10.13.0/lib/node_modules/renovate/dist/platform/git/storage.js:129:23)\n    at process._tickCallback (internal/process/next_tick.js:68:7)"
       }
 INFO: Unknown GitLab initRepo error (repository=groupName/projectName)
       "err": {
         "message": "platform-failure",
         "stack": "Error: platform-failure\n    at Storage.initRepo (/Users/foo/.nvm/versions/node/v10.13.0/lib/node_modules/renovate/dist/platform/git/storage.js:129:23)\n    at process._tickCallback (internal/process/next_tick.js:68:7)"
       }
 INFO: Platform error - skipping (repository=groupName/projectName)
```

**Describe the solution you'd like**

Include this filter in the `initRepo()` to log an exception but handle it such that `autodiscover` doesn't fail fatally.

**Describe alternatives you've considered**

Write a script that generates an explicit list of projects that have the repo enabled and then inject that into the `repositories` config value.  This clearly would work but seems to be a less-than-ideal hack or workaround for this functionality.

**Comment by arsdehnel** [comment]
I was thinking about trying to make this change myself but wanted to make sure this seemed to match what what was desired for the tool.  Looks like this might be about a 3-line fix at https://github.com/renovatebot/renovate/blob/master/lib/platform/gitlab/index.ts#L140 that would be 

```ts
    if (res.body. repository_access_level === "disabled") {
      logger.info(
        'Repository portion of project is disabled - throwing error to abort renovation'
      );
      throw new Error('disabled');
    }
```

**Comment by rarkins** [comment]
Yes, that should solve it

**Comment by JamieMagee** [comment]
Closed by #5084
